packages/web-react/src/components/Button/useButtonAriaProps.tsx

```diff
@@ -0,0 +1,54 @@
+import { AnchorHTMLAttributes, ButtonHTMLAttributes, ElementType, HTMLAttributes } from 'react';
+import { ClickEvent, AriaButtonProps } from '../../types';
+
+export interface ButtonAria<T> {
+  /** Props for the button element. */
+  buttonProps: T;
+}
+
+export function useButtonAriaProps(props: AriaButtonProps<'a'>): ButtonAria<AnchorHTMLAttributes<HTMLAnchorElement>>;
```

Comment:
I was heavily inspiried by https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/button/src/useButton.ts to deal with this problem of two types of button. Name it `aria` seems to me right because this deal only with real `button` or `a` attributes.

Also, we are not dealing with all aria props at the moment but I think this is a starting point. So I think that naming it right from the beginning would be appropriate.